### PR TITLE
MUMUP-2102 Loading state flag feature for generic widgets

### PR DESCRIPTION
**Pull request by jiayinjx**

### CODE REVIEW

File: angularjs-portal-home/src/main/webapp/my-app/layout/widget/partials/widget-card.html [hunk 1]
```diff
@@ -29,25 +29,28 @@ <h4 id="appTitle_portlet.title-{{::portlet.nodeId}}" aria-labelledby="appTitle_p
...
   <div ng-if="'RSS' === widgetCtrl.portletType(portlet)">
     <rss app="portlet" config="portlet.widgetConfig"></rss>
   </div>
-  
+
   <div ng-if="'GENERIC' === widgetCtrl.portletType(portlet)">
     <div ng-controller="GenericWidgetController as genericWidgetCtrl">
+        <div ng-if="loading" id="loading">
+            <loading-gif data-object='content' data-empty='isEmpty'></loading-gif>
+        </div>
```
Comment (timlevett) [inline angularjs-portal-home/src/main/webapp/my-app/layout/widget/partials/widget-card.html]: The original idea of the "loading state" was that the widget would have the option to have a loading state. I do like this uniformed approach tho. Makes way more sense.

File: angularjs-portal-home/src/main/webapp/my-app/layout/widget/partials/widget-card.html [hunk 2]
```diff
@@ -29,25 +29,28 @@ <h4 id="appTitle_portlet.title-{{::portlet.nodeId}}" aria-labelledby="appTitle_p
...
         <content-item></content-item>
```
Comment (timlevett) [inline angularjs-portal-home/src/main/webapp/my-app/layout/widget/partials/widget-card.html]: Should we add in a `ng-if='!loading'` here?

Comment (jiayinjx) [inline angularjs-portal-home/src/main/webapp/my-app/layout/widget/partials/widget-card.html]: I think we need it, when I try to remove it and run, the loading-gif will
be there forever.

Comment (timlevett) [inline angularjs-portal-home/src/main/webapp/my-app/layout/widget/partials/widget-card.html]: I was commenting saying we should add the !loading to content-item. So during loading it doesn't show the content-item content.

Comment (jiayinjx) [inline angularjs-portal-home/src/main/webapp/my-app/layout/widget/partials/widget-card.html]: got it, update just now.

**Comment by timlevett** [comment]
:+1:

**Pull request merged**